spyder/widgets/sourcecode/codeeditor.py

```diff
@@ -2206,14 +2206,15 @@ def toggle_comment(self):
 
     def comment(self):
         """Comment current line or selection."""
-        self.add_prefix(self.comment_string)
+        self.add_prefix(self.comment_string + " ")
 
     def uncomment(self):
         """Uncomment current line or selection."""
         self.remove_prefix(self.comment_string)
+        self.remove_prefix(" ")
 
     def __blockcomment_bar(self):
-        return self.comment_string + '='*(79-len(self.comment_string))
+        return self.comment_string + ' ' + '=' * (78 - len(self.comment_string))

```

Comment:
It is because of the space added between the comment string and the '='. Maybe this could be rewritten as:

```python
return self.comment_string + ' ' + '=' * (79 - len(self.comment_string + ' '))
```